homework/src/index.js

```diff
@@ -30,18 +33,95 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
-    });
+  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  const root = document.getElementById('root');
+  createAndAppend("img", root, { src: "./hyf.png", id: "imgLogo"});
+  const divSelect = createAndAppend("header", root);
+  createAndAppend("p", divSelect, { html: "HYF Repositories : ", id: "pContributors", role: "Heading" });
+  const repoSelect = createAndAppend("select", divSelect, {role: "listbox"});
+  const divFlex = createAndAppend("section", root);
+  const divInfo = createAndAppend("article", divFlex, { id: "divInfo" , role: "article"});
+  const divCont = createAndAppend("article", divFlex, { id: "divCont" , role: "article" });
+
+  async function main(HYF_REPOS_URL) {
+    try {
+      const repositories = await fetchJSON(HYF_REPOS_URL);
+
+      repositories.forEach((repository, index) => {
+        createAndAppend("option", repoSelect, { html: repository.name, value: index, role: "option" });
+      });
+      repoSelect.addEventListener("change", (event) => {
+        const optionIndex = event.target.value;
+        renderRepository(divInfo, repositories[optionIndex]);
+        renderContributors(divCont, repositories[optionIndex]);
+      });
+      const initialRender = repositories[0];
+      renderRepository(divInfo, initialRender);
+      renderContributors(divCont, initialRender);
+    }
+
+    catch (error) {
+      error => {
+        createAndAppend('div', root, { html: error.message, class: 'alert-error', role: "alert" });
+        return;
+      };
```

Comment:
Your error handler doesn't work. Have you tried it?

Reply:
I corrected code. I had not noticed this error before i uploaded it